presto-jdbc/src/main/java/io/prestosql/jdbc/ConnectionProperties.java

```diff
@@ -378,4 +380,42 @@ public ExtraCredentials()
             return ImmutableList.of(name, value);
         }
     }
+
+    private static class SessionProperties
+            extends AbstractConnectionProperty<Map<String, String>>
+    {
+        private static final Splitter NAME_PARTS_SPLITTER = Splitter.on('.');
+
+        public SessionProperties()
+        {
+            super("sessionProperties", NOT_REQUIRED, ALLOWED, SessionProperties::parseSessionProperties);
+        }
+
+        // Session properties consists of a list of session property name value pairs.
+        // E.g., `jdbc:presto://example.net:8080/?sessionProperties=abc:xyz;catalog.foo:bar` will create session properties `abc=xyz` and `catalog.foo=bar`
+        public static Map<String, String> parseSessionProperties(String sessionProperties)
+        {
+            return MAP_ENTRIES_SPLITTER.splitToList(sessionProperties).stream()
+                    .map(SessionProperties::parseSessionProperty)
+                    .collect(toImmutableMap(entry -> entry.get(0), entry -> entry.get(1)));
+        }
+
+        public static List<String> parseSessionProperty(String sessionProperty)
+        {
+            List<String> nameValue = MAP_ENTRY_SPLITTER.splitToList(sessionProperty);
+            checkArgument(nameValue.size() == 2, "Malformed session property: %s", sessionProperty);
+
+            String name = nameValue.get(0);
+            checkArgument(!name.isEmpty(), "Session property name is empty");
+            checkArgument(NAME_PARTS_SPLITTER.splitToList(name).size() <= 2, "Malformed session property name: %s", name);
```

Comment:
Can catalog contain a dot or a space?
eg maybe this is a catalog which would require quoting in SQL?

(For now it's sufficient to validate `name` does not contain `"` so that we are future-proof)